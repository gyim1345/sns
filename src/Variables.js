export const post = {
  like: [('guy@gmail.com', 'noone@gmail.com', 'gibong@gmail.com')],
  tag: [('boat', 'sky', 'landscape')],
  _id: '5e8ec52d9bb4913b5a844423',
  id: 1,
  title: 'posting with id 1',
  imageUrl: 'http://localhost:3000/static/images/1.jpg',
  userName: 'gibong@gmail.com'
};

export const posts = [
  {
    like: [('guy@gmail.com', 'noone@gmail.com', 'gibong@gmail.com')],
    tag: [('boat', 'sky', 'landscape')],
    _id: '5e8ec52d9bb4913b5a844423',
    id: 1,
    title: 'posting with id 1',
    imageUrl: 'http://localhost:3000/static/images/1.jpg',
    userName: 'gibong@gmail.com'
  }
];

export const currentUser = 'gibong@gmail.com';

export const currentUserName = 'gibong'

export const comments = [
  {
    like: [('gibong@gmail.com', 'guy@gmail.com', 'noone@gmail.com')],
    _id: '5e90119435a2d94f9e08e35a',
    id: 1,
    postLId: 1,
    title: 'title1',
    userName: 'gibong@gmail.com',
    replyToCommentId: null
  }
];

export const info = {
  userURL: 'http://localhost:3000/static/images/user1.png',
  name: 'gibong@gmail.com',
  postCount: 16,
  followerNumber: 3,
  nickName: '기봉이',
  introductory: '나는 기봉이'
};

export const sizeOfPicture = {
  width: '10px',
  height: '10px'
};
