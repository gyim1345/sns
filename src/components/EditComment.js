import React, { useState } from 'react';
import Swal from 'sweetalert2';

import { editCommentAPI } from '../apis/comment';

function EditComment({
  posting,
  setPosting,
  indexOfCommentOnThisPosting,
  currentUser,
  setCommentAPI
}) {
  const thisPost = posting[indexOfCommentOnThisPosting] || posting;
  const [input, setInput] = useState([thisPost.title]);

  const onEdit = e => {
    setInput(e.target.value);
  };

  const onClick = async () => {
    try {
      const response = await editCommentAPI(
        input,
        posting,
        currentUser,
        indexOfCommentOnThisPosting
      );
      response.Message !== undefined
        ? Swal.fire({
            icon: 'error',
            title: 'Oops...',
            text: `${response.Message}`
          })
        : indexOfCommentOnThisPosting === undefined
        ? setPosting(response)
        : setCommentAPI(response);
    } catch (e) {
      Swal.fire({
        icon: 'error',
        title: 'Oops...',
        text: 'Internal Error'
      });
    }
  };

  return (
    <>
      <input type="text" value={input} onChange={e => onEdit(e)} />
      <button type="button" onClick={onClick} id="buttonEdit">
        Edit
      </button>
    </>
  );
}

// Edit.propTypes = {
//   currentUser: PropTypes.string,
//   posting: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
//   setPosting: PropTypes.elementType,
//   indexOfCommentOnThisPosting: PropTypes.number,
//   setCommentAPI: PropTypes.elementType
// };

// Edit.defaultProps = {
//   currentUser: "",
//   posting: {},
//   indexOfComment: undefined,
//   cid: undefined
// };

export default EditComment;
