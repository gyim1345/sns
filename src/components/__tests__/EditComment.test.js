import React from 'react';
import { mount } from 'enzyme';
import EditComment from '../EditComment';
import { comments, currentUser } from '../../Variables';

describe('<EditComment />', () => {
  it('스냅샷 비교', () => {
    const wrapper = mount(
      <EditComment
        posting={comments}
        indexOfCommentOnThisPosting={0}
        currentUser={currentUser}
      />
    );

    expect(wrapper.props().posting).toBe(comments);
    expect(wrapper.props().indexOfCommentOnThisPosting).toBe(0);
    expect(wrapper.props().currentUser).toBe(currentUser);
    expect(wrapper.html()).toMatch('<input');
    expect(wrapper.html()).toMatch('<button');
    expect(wrapper.text()).toMatch('Edit');
    expect(wrapper).toMatchSnapshot();
  });
});
