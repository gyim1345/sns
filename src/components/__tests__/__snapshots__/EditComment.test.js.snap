// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditComment /> 스냅샷 비교 1`] = `
<EditComment
  currentUser="gibong@gmail.com"
  indexOfCommentOnThisPosting={0}
  posting={
    Array [
      Object {
        "_id": "5e90119435a2d94f9e08e35a",
        "id": 1,
        "like": Array [
          "noone@gmail.com",
        ],
        "postLId": 1,
        "replyToCommentId": null,
        "title": "title1",
        "userName": "gibong@gmail.com",
      },
    ]
  }
>
  <input
    onChange={[Function]}
    type="text"
    value={
      Array [
        "title1",
      ]
    }
  />
  <button
    id="buttonEdit"
    onClick={[Function]}
    type="button"
  >
    Edit
  </button>
</EditComment>
`;
